service/common/src/main/java/org/apache/polaris/service/admin/PolarisAdminService.java

```diff
@@ -664,6 +665,15 @@ private void validateUpdateCatalogDiffOrThrow(
             "Cannot modify ExternalId in storage config from %s to %s",
             currentStorageConfig, newStorageConfig);
       }
+
+      if ((currentAwsConfig.getUserARN() != null
+              && !currentAwsConfig.getUserARN().equals(newAwsConfig.getUserARN()))
+          || (newAwsConfig.getUserARN() != null
+              && !newAwsConfig.getUserARN().equals(currentAwsConfig.getUserARN()))) {
+        throw new BadRequestException(
+            "Cannot modify userARN in storage config from %s to %s",
```

Comment:
Why?

A Polaris Admin should have control over any configuration. `userARN` is ultimately controlled by the owner of the catalog and corresponding storage. Why would Polaris code prevent changes to these values?

Reply:
`userARN` represents the polaris service, from my understanding, it should be configured by the polaris service provider rather than the user.

The owner of the catalog is an user, user only needs to provide an IAM role, polaris service will use its AWS credential (the `userARN`) to assume the role and get the temp AWS credential. How could an user provide an AWS user ARN for polaris service?

Here is the guidance from Open Catalog for creating a catalog: https://other-docs.snowflake.com/en/opencatalog/create-catalog

Reply:
> userARN represents the polaris service [...]

Then it should not be stored in catalog properties at all, IMHO.

Reply:
I think this opens a can of worms :)

From the OSS perspective I can imagine users of Apache Polaris, who may want to have one service, but use different `userArn` for different catalogs. 

Ultimately, it depends on the scope of control the catalog owner has and the service owner has. Deployment scenarios can vary vastly.

I'd propose to cover this by a feature flag (using service-level `userArn` or catalog-level)... but it might be best to discuss this on the `dev` ML.

Reply:
Thanks Dmitri! Creating one `userArn` for each catalog is very expensive (an user arn means a real AWS user and a AWS credential), instead of relying on different `userArn`, users should use `externalID` if they only want their IAM role to be assumed by polaris for a specific catalog. 
I will open a thread in `dev` ML, let's discuss on the thread!

Reply:
From an Apache Polaris user perspective, I do not really see how `userArn` affects the behaviour of Polaris (see also my more specific comment below).

Overall, it looks like this area of functionality may need a review :thinking: